### PR TITLE
[ja] Translate docs/concepts/storage/ephemeral-volumes/ into Japanese

### LINKED ISSUE #38065: [ja] Translate `docs/concepts/storage/ephemeral-volumes/` into Japanese
**This is a Feature Request**

**What would you like to be added**

Translate https://kubernetes.io/docs/concepts/storage/ephemeral-volumes/ into Japanese
**Why is this needed**
This page has not been translated.
**Comments**

/language ja
/assign

### PR DESCRIPTION
**Pull request by KobayashiD27**
close #38065

### CODE REVIEW

File: content/ja/docs/concepts/storage/ephemeral-volumes.md [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+ボリュームはPodの存続期間に従い、Podとともに作成および削除されるため、Podは、永続ボリュームが利用可能な場所に制限されることなく停止および再起動できます。
+
+エフェメラルボリュームはPod仕様で*インライン*で指定されているため、アプリケーションの展開と管理が簡素化されます。
+
+### エフェメラルボリュームのタイプ {#types-of-ephemeral-volumes}
+
+Kubernetesは、さまざまな目的のためにいくつかの異なる種類のエフェメラルボリュームをサポートしています。
+- [emptyDir](/ja/docs/concepts/storage/volumes/#emptydir):Podの起動時には空で、ストレージはkubeletベースディレクトリ(通常はルートディスク)またはRAMからローカルに取得されます。
+- [configMap](/ja/docs/concepts/storage/volumes/#configmap)、[downwardAPI](/ja/docs/concepts/storage/volumes/#downwardapi)、[secret](/ja/docs/concepts/storage/volumes/#secret):Podにさまざまな種類のKubernetesデータを挿入します。
+- [CSI ephemeral volumes](#csi-ephemeral-volumes):上のボリュームの種類に似ていますが、特に[この機能をサポートする](https://kubernetes-csi.github.io/docs/drivers.html)特別な[CSIドライバー](https://github.com/container-storage-interface/spec/blob/master/spec.md)によって提供されます。
```
Comment (nasa9084) [inline content/ja/docs/concepts/storage/ephemeral-volumes.md:28]:
```suggestion
- [CSIエフェメラルボリューム](#csi-ephemeral-volumes):上のボリュームの種類に似ていますが、特に[この機能をサポートする](https://kubernetes-csi.github.io/docs/drivers.html)特別な[CSIドライバー](https://github.com/container-storage-interface/spec/blob/master/spec.md)によって提供されます。
```

File: content/ja/docs/concepts/storage/ephemeral-volumes.md [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+- [generic ephemeral volumes](#generic-ephemeral-volumes):これは、永続ボリュームもサポートするすべてのストレージドライバーで提供できます。
```
Comment (nasa9084) [inline content/ja/docs/concepts/storage/ephemeral-volumes.md:29]:
```suggestion
- [汎用エフェメラルボリューム](#generic-ephemeral-volumes):これは、永続ボリュームもサポートするすべてのストレージドライバーで提供できます。
```

File: content/ja/docs/concepts/storage/ephemeral-volumes.md [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+
+`emptyDir`、`configMap`、`downwardAPI`、`secret`は[ローカルエフェメラルストレージ](/ja/docs/concepts/configuration/manage-resources-containers/#local-ephemeral-storage)として提供されます。
+これらは、各ノードのkubeletによって管理されます。
+
+CSIエフェメラルボリュームは、サードパーティのCSIストレージドライバーによって提供される必要があります。
```
Comment (nasa9084) [inline content/ja/docs/concepts/storage/ephemeral-volumes.md:34]:
```suggestion
CSIエフェメラルボリュームは、サードパーティーのCSIストレージドライバーによって提供される必要があります。
```

File: content/ja/docs/concepts/storage/ephemeral-volumes.md [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+
+汎用エフェメラルボリュームは、サードパーティのCSIストレージドライバーによって提供される可能性がありますが、動的プロビジョニングをサポートする他のストレージドライバーによって提供されることもあります。一部のCSIドライバーは、CSIエフェメラルボリューム用に特別に作成されており、動的プロビジョニングをサポートしていません。これらは汎用エフェメラルボリュームには使用できません。
```
Comment (nasa9084) [inline content/ja/docs/concepts/storage/ephemeral-volumes.md:36]:
```suggestion
汎用エフェメラルボリュームは、サードパーティーのCSIストレージドライバーによって提供される可能性がありますが、動的プロビジョニングをサポートする他のストレージドライバーによって提供されることもあります。一部のCSIドライバーは、CSIエフェメラルボリューム用に特別に作成されており、動的プロビジョニングをサポートしていません。これらは汎用エフェメラルボリュームには使用できません。
```

File: content/ja/docs/concepts/storage/ephemeral-volumes.md [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+
+サードパーティ製ドライバーを使用する利点は、Kubernetes自体がサポートしていない機能を提供できることです。たとえば、kubeletによって管理されるディスクとは異なるパフォーマンス特性を持つストレージや、異なるデータの挿入などです。
```
Comment (nasa9084) [inline content/ja/docs/concepts/storage/ephemeral-volumes.md:38]:
```suggestion
サードパーティー製ドライバーを使用する利点は、Kubernetes自体がサポートしていない機能を提供できることです。たとえば、kubeletによって管理されるディスクとは異なるパフォーマンス特性を持つストレージや、異なるデータの挿入などです。
```

File: content/ja/docs/concepts/storage/ephemeral-volumes.md [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+
+`volumeAttributes`は、ドライバーによって準備されるボリュームを決定します。これらの属性は各ドライバーに固有のものであり、標準化されていません。詳細な手順については、各CSIドライバーのドキュメントを参照してください。
+
+### CSIドライバーの制限事項 {#csi-driver-restrictions}
+
+CSIエフェメラルボリュームを使用すると、ユーザーはPod仕様の一部として`volumeAttributes`をCSIドライバーに直接提供できます。
+通常は管理者に制限されている`volumeAttributes`を許可するCSIドライバーは、インラインエフェメラルボリュームでの使用には適していません。
+たとえば、通常StorageClassで定義されるパラメーターは、インラインエフェメラルボリュームを使用してユーザーに公開しないでください。
+
+ポッド仕様内でインラインボリュームとして使用できるCSIドライバーを制限する必要があるクラスタ管理者は、次の方法で行うことができます。
```
Comment (nasa9084) [inline content/ja/docs/concepts/storage/ephemeral-volumes.md:86]:
```suggestion
Pod仕様内でインラインボリュームとして使用できるCSIドライバーを制限する必要があるクラスター管理者は、次の方法で行うことができます。
```

File: content/ja/docs/concepts/storage/ephemeral-volumes.md [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+
+- CSIドライバー仕様の`volumeLifecycleModes`から`Ephemeral`を削除します。これにより、ドライバーをインラインエフェメラルボリュームとして使用できなくなります。
+- [admission webhook](/docs/reference/access-authn-authz/extensible-admission-controllers/)を使用して、このドライバーの使用方法を制限します。
+
+### 汎用エフェメラルボリューム {#generic-ephemeral-volumes}
+
+{{< feature-state for_k8s_version="v1.23" state="stable" >}}
+
+汎用エフェメラルボリュームは、プロビジョニング後に通常は空であるスクラッチデータ用のポッドごとのディレクトリを提供するという意味で、`emptyDir`ボリュームに似ています。ただし、追加の機能がある場合もあります。
```
Comment (nasa9084) [inline content/ja/docs/concepts/storage/ephemeral-volumes.md:95]:
```suggestion
汎用エフェメラルボリュームは、プロビジョニング後に通常は空であるスクラッチデータ用のPodごとのディレクトリを提供するという意味で、`emptyDir`ボリュームに似ています。ただし、追加の機能がある場合もあります。
```

File: content/ja/docs/concepts/storage/ephemeral-volumes.md [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+            accessModes: [ "ReadWriteOnce" ]
+            storageClassName: "scratch-storage-class"
+            resources:
+              requests:
+                storage: 1Gi
+```
+
+### LifecycleとPersistentVolumeClaim {#lifecycle-and-persistentvolumeclaim}
+
+設計上の重要なアイデアは、[ボリュームクレームのパラメータ](/docs/reference/generated/kubernetes-api/{{< param "version" >}}/#ephemeralvolumesource-v1alpha1-core)がPodのボリュームソース内で許可されることです。
```
Comment (nasa9084) [inline content/ja/docs/concepts/storage/ephemeral-volumes.md:134]:
```suggestion
設計上の重要なアイデアは、[ボリュームクレームのパラメーター](/docs/reference/generated/kubernetes-api/{{< param "version" >}}/#ephemeralvolumesource-v1alpha1-core)がPodのボリュームソース内で許可されることです。
```

File: content/ja/docs/concepts/storage/ephemeral-volumes.md [hunk 9]
```diff
@@ -0,0 +1,178 @@
...
+PersistentVolumeClaimのラベル、アノテーション、および一連のフィールド全体がサポートされています。
+そのようなPodが作成されると、エフェメラルボリュームコントローラーは、Podと同じ名前空間に実際のPersistentVolumeClaimオブジェクトを作成し、Podが削除されたときにPersistentVolumeClaimが確実に削除されるようにします。
+
+これにより、ボリュームバインディングおよび/またはプロビジョニングがトリガーされます。
+これは、{{< glossary_tooltip text="StorageClass" term_id="storage-class" >}}が即時ボリュームバインディングを使用する場合、またはPodが一時的にノードにスケジュールされている場合(`WaitForFirstConsumer`ボリュームバインディングモード)のいずれかです。
+後者は、スケジューラがPodに適したノードを自由に選択できるため、一般的なエフェメラルボリュームに推奨されます。即時バインディングでは、ボリュームが利用可能になった時点で、ボリュームにアクセスできるノードをスケジューラが選択する必要があります。
```
Comment (nasa9084) [inline content/ja/docs/concepts/storage/ephemeral-volumes.md:140]:
```suggestion
後者は、スケジューラーがPodに適したノードを自由に選択できるため、一般的なエフェメラルボリュームに推奨されます。即時バインディングでは、ボリュームが利用可能になった時点で、ボリュームにアクセスできるノードをスケジューラーが選択する必要があります。
```

File: content/ja/docs/concepts/storage/ephemeral-volumes.md [hunk 10]
```diff
@@ -0,0 +1,178 @@
...
+### PersistentVolumeClaimの命名 {#persistentpolumeplaim-naming}
+
+自動的に作成されたPVCの命名は決定論的です。名前はPod名とボリューム名を組み合わせたもので、途中にハイフン(`-`)があります。上記の例では、PVC名は`my-app-scratch-volume`になります。この決定論的な命名により、Pod名とボリューム名が分かればPVCを検索する必要がないため、PVCとの対話が容易になります。
+
+また、決定論的な命名では、異なるPod間、およびPodと手動で作成されたPVCの間で競合が発生する可能性があります(ボリュームが"scratch"のPod"pod-a"と、名前が"pod"でボリュームが"a-scratch"の別のPodは、どちらも同じPVC名"pod-a-scratch")。
+
+次のような競合が検出されます。Pod用に作成された場合、PVCはエフェメラルボリュームにのみ使用されます。このチェックは、所有関係に基づいています。既存のPVCは上書きまたは変更されません。ただし、適切なPVCがないとPodを起動できないため、これでは競合が解決されません。
+
+{{< caution >}}
+これらの競合が発生しないように、同じ名前空間内でポッドとボリュームに名前を付けるときは注意してください。
```
Comment (nasa9084) [inline content/ja/docs/concepts/storage/ephemeral-volumes.md:155]:
```suggestion
これらの競合が発生しないように、同じ名前空間内でPodとボリュームに名前を付けるときは注意してください。
```

File: content/ja/docs/concepts/storage/ephemeral-volumes.md [hunk 11]
```diff
@@ -0,0 +1,178 @@
...
+{{< /caution >}}
+
+### セキュリティ {#security}
+
+GenericEphemeralVolume機能を有効にすると、ユーザーは、PVCを直接作成する権限がなくても、Podを作成できる場合、間接的にPVCを作成できます。クラスタ管理者はこれを認識している必要があります。これがセキュリティモデルに適合しない場合は、一般的なエフェメラルボリュームを持つPodなどのオブジェクトを拒否する[admission webhook](/docs/reference/access-authn-authz/extensible-admission-controllers/)を使用する必要があります。
```
Comment (nasa9084) [inline content/ja/docs/concepts/storage/ephemeral-volumes.md:160]:
```suggestion
GenericEphemeralVolume機能を有効にすると、ユーザーは、PVCを直接作成する権限がなくても、Podを作成できる場合、間接的にPVCを作成できます。クラスター管理者はこれを認識している必要があります。これがセキュリティモデルに適合しない場合は、一般的なエフェメラルボリュームを持つPodなどのオブジェクトを拒否する[admission webhook](/docs/reference/access-authn-authz/extensible-admission-controllers/)を使用する必要があります。
```

**Review by nasa9084 — COMMENTED** [review]
/assign

File: content/ja/docs/concepts/storage/ephemeral-volumes.md (hunk 11 above)
Comment (KobayashiD27) [reply content/ja/docs/concepts/storage/ephemeral-volumes.md]: Thank you for your review. I've updated them.

File: content/ja/docs/concepts/storage/ephemeral-volumes.md [hunk 12]
```diff
@@ -0,0 +1,178 @@
...
+
+{{< note >}}
+CSIエフェメラルボリュームは、CSIドライバーのサブセットによってのみサポートされます。
+Kubernetes CSI[ドライバーリスト](https://kubernetes-csi.github.io/docs/drivers.html)には、エフェメラルボリュームをサポートするドライバーが表示されます。
+{{< /note >}}
+
+概念的には、CSIエフェメラルボリュームは`configMap`、`downwardAPI`、および`secret`ボリュームタイプに似ています。
+ストレージは各ノードでローカルに管理され、Podがノードにスケジュールされた後に他のローカルリソースと一緒に作成されます。Kubernetesには、この段階でPodを再スケジュールするという概念はもうありません。
+ボリュームの作成は、失敗する可能性が低くなければなりません。さもないと、Podの起動が停止します。
+特に、[ストレージ容量を考慮したPodスケジューリング](/ja/docs/concepts/storage/storage-capacity/)は、これらのボリュームではサポートされていません。
```
Comment (t-inu) [inline content/ja/docs/concepts/storage/ephemeral-volumes.md:52]:
```suggestion
特に、[ストレージ容量を考慮したPodスケジューリング](/ja/docs/concepts/storage/storage-capacity/)は、これらのボリュームではサポートされて*いません*。
```
> `[storage capacity aware Pod scheduling](/docs/concepts/storage/storage-capacity/) is *not* supported ...`

File: content/ja/docs/concepts/storage/ephemeral-volumes.md [hunk 13]
```diff
@@ -0,0 +1,178 @@
...
+### 汎用エフェメラルボリューム {#generic-ephemeral-volumes}
+
+{{< feature-state for_k8s_version="v1.23" state="stable" >}}
+
+汎用エフェメラルボリュームは、プロビジョニング後に通常は空であるスクラッチデータ用のPodごとのディレクトリを提供するという意味で、`emptyDir`ボリュームに似ています。ただし、追加の機能がある場合もあります。
+
+- ストレージは、ローカルまたはネットワークに接続できます。
+- ボリュームは、Podが超えることができない固定サイズを持つことができます。
+- ボリュームには、ドライバーとパラメーターによっては、いくつかの初期データがある場合があります。
+- [スナップショット](/docs/concepts/storage/volume-snapshots/),[クローン作成](/ja/docs/concepts/storage/volume-pvc-datasource/),[サイズ変更](/ja/docs/concepts/storage/persistent-volumes/#expanding-persistent-volumes-claims),[ストレージ容量の追跡](/ja/docs/concepts/storage/storage-capacity/)などボリュームに対する一般的な操作は、ドライバーがそれらをサポートしていることを前提としてサポートされています。
```
Comment (t-inu) [inline content/ja/docs/concepts/storage/ephemeral-volumes.md:100]:
```suggestion
- [スナップショット](/docs/concepts/storage/volume-snapshots/)、[クローン作成](/ja/docs/concepts/storage/volume-pvc-datasource/)、[サイズ変更](/ja/docs/concepts/storage/persistent-volumes/#expanding-persistent-volumes-claims)、[ストレージ容量の追跡](/ja/docs/concepts/storage/storage-capacity/)などボリュームに対する一般的な操作は、ドライバーがそれらをサポートしていることを前提としてサポートされています。
```

**Review by t-inu — APPROVED** [review]
/lgtm

**Pull request merged**